ctl/src/main.rs

```diff
@@ -48,13 +50,13 @@ fn main() {
   match matches.cmd {
     SubCmd::Shutdown{ hard, worker} => {
       if hard {
-        hard_stop(channel, worker, timeout);
+        hard_stop(channel, worker, timeout)
       } else {
-        soft_stop(channel, worker);
+        soft_stop(channel, worker)
       }
     },
     SubCmd::Upgrade { worker: None } => upgrade_main(channel, &config),
-    SubCmd::Upgrade { worker: Some(id) } => { upgrade_worker(channel, timeout, id); },
+    SubCmd::Upgrade { worker: Some(id) } => { upgrade_worker(channel, timeout, id)?; Ok(()) },
```

Comment:
You should add a line break here

Reply:
Is `upgrade_worker` function returning something? If it returns a Unit, you could write this instead:
```rust
Ok(upgrade_worker(channel, timeout, id)?)
```

Reply:
`upgrade_worker()` does return `Ok(Channel<CommandRequest, CommandResponse>)`, so I couldn't wrap it so easily. 

I added the line break in a follow-up commit.